client/app/components/queries/QueryEditor/AutoLimitCheckbox.jsx

```diff
@@ -0,0 +1,28 @@
+import React, { useCallback } from "react";
+import PropTypes from "prop-types";
+import "@/redash-font/style.less";
+import recordEvent from "@/services/recordEvent";
+import { Checkbox } from "antd";
+
+export default function AutoLimitCheckbox({ available, checked, onChange }) {
+  const handleClick = useCallback(() => {
+    recordEvent("checkbox_auto_limit", "screen", "query_editor", { state: !checked });
+    onChange(!checked);
+  }, [checked, onChange]);
+
+  return (
+    <Checkbox
+      className="query-editor-controls-checkbox m-l-5"
+      disabled={!available}
```

Comment:
I wonder if it's better to just hide this checkbox in UI when the feature is not available instead of disabling it 🤔

I don't see a reason to keep it and it also doesn't self-explain on "why this is disabled"

Reply:
We checked with Wenbo on this, and he agrees that we can just disable it instead of just hide it,

Reply:
WDYT about adding a tooltip with something like "Auto limiting is not available for this data source"?

Reply:
Just added!